re2c: 1.0.3 -> 1.2.1

trofi opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

Commit:
re2c: 1.0.3 -> 1.2.1

thoughtpolice reviewed (changes requested):
Thank you! Just one minor nit: please rebase this on top of the `staging` branch instead of `master`. This will introduce a delay (few days to ~2 weeks, IME) before you'll see `1.2.1` in `master`, but will save people a lot of grief by avoiding a lot of rebuilds. That's because `re2c` has some pretty deep transitive dependencies across the source tree, including `stdenv`.

A good rule of thumb (for me) is that if you're looking at ~1000 rebuilds, you might want to ask someone -- a lot of the decision can depend on the changes in question. But at ~2000 rebuilds or more in total, it should definitely go to staging. Our bot @GrahamcOfBorg will add some tags indicating how many rebuilds it estimates will happen, so you can get an idea of where it should go.

This change otherwise looks just fine -- thanks! (I'm only asking for 'request changes' to ensure someone doesn't accidentally hit the merge button before this is rebased.)

trofi commented:
The large rebuild impact makes sense. Rebased on to of `staging`.

Commit:
re2c: enable parallel building, enable tests

Parallel builds speed up building considerably.
Tests don't add much of an overhead.

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

trofi commented:
While at it enabled parallel builds and tests.
